src/rgw/rgw_rest_s3.cc

```diff
@@ -462,6 +467,33 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs,
       i != attrs.end()) {
     dump_header(s, "x-amz-replication-status", i->second);
   }
+  // restore status
+  if (attr_iter != attrs.end()) {
+    bufferlist bl = attr_iter->second;
+    auto iter = bl.cbegin();
+    decode(restore_status, iter);
+  }
+
+  if (expire_iter != attrs.end()) {
+    bufferlist bl = expire_iter->second;
+    auto iter = bl.cbegin();
+    decode(expiration_date, iter);
+  }
+
+  if (attr_iter == attrs.end() || restore_status == rgw::sal::RGWRestoreStatus::RestoreFailed) {
+    s->err.http_ret = 202; //Accepted
+    dump_header(s, "x-amz-restore", "on-going-request=\"true\"");
```

Comment:
should we treat restore failure as error than 202? And the on-going-request should be true only for restoreinprogress scenario, all other cases it should be false, right?